arc_cache.go

```diff
@@ -112,15 +152,14 @@ func (b *arccache) GetSize(ctx context.Context, k cid.Cid) (int, error) {
 		// we have it but don't know the size, ask the datastore.
 	}
 
-	lk := b.getLock(k)
-	lk.RLock()
-	defer lk.RUnlock()
+	b.lock(key, false)
+	defer b.unlock(key, false)
```

Comment:
Maybe it's not worthwhile/sufficiently rare, but is it worthwhile to be checking the cache again or taking the read lock earlier? Otherwise you're going to hit the underlying storage twice if you get to requests at a similar time.

It's probably fine to say that that pattern is sufficiently rare as to not be worth the extra work for other block reads though.

Reply:
We could take the read lock earlier, but I'd like to avoid the overhead in case the item is in the cache. I'd just optimize this later if it's an issue.